File: app/src/main/java/com/mdmobile/cyclops/dataTypes/ProfileActions.java
package com.mdmobile.cyclops.dataTypes;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.StringDef;

import static com.mdmobile.cyclops.dataTypes.ProfileActions.INSTALL;
import static com.mdmobile.cyclops.dataTypes.ProfileActions.REVOKE;

/**
 * Constants to define possible api actions
 */
@SuppressWarnings("unused")
@StringDef({INSTALL, REVOKE})
@Retention(RetentionPolicy.SOURCE)
public @interface ProfileActions {
    String INSTALL = "install";
    String REVOKE = "revoke";
}

File: app/src/main/java/com/mdmobile/cyclops/dataTypes/ProfileActions.kt
package com.mdmobile.cyclops.dataTypes

import androidx.annotation.StringDef
import com.mdmobile.cyclops.dataTypes.ProfileActions.Companion.INSTALL
import com.mdmobile.cyclops.dataTypes.ProfileActions.Companion.REVOKE

/**
 * Constants to define possible api actions
 */
@StringDef(INSTALL, REVOKE)
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class ProfileActions {
    companion object {
        const val INSTALL = "install"
        const val REVOKE = "revoke"
    }
}

